Mathlib/Algebra/Order/Module/Defs.lean

```diff
@@ -794,6 +794,47 @@ instance instSMulPosReflectLE [SMulPosReflectLE α β] : SMulPosReflectLE α β
 end Right
 end OrderDual
 
+section OrderedAddCommMonoid
+variable [StrictOrderedSemiring α] [ExistsAddOfLE α] [OrderedCancelAddCommMonoid β]
+  [Module α β]
+
+section PosSMulMono
+variable [PosSMulMono α β] {a₁ a₂ : α} {b₁ b₂ : β}
+
+/-- Binary **rearrangement inequality**. -/
+lemma smul_add_smul_le_smul_add_smul (ha : a₁ ≤ a₂) (hb : b₁ ≤ b₂) :
+    a₁ • b₂ + a₂ • b₁ ≤ a₁ • b₁ + a₂ • b₂ := by
+  obtain ⟨a, ha₀, rfl⟩ := exists_nonneg_add_of_le ha
+  rw [add_smul, add_smul, add_left_comm]
+  gcongr
+  exact ha₀
+
+/-- Binary **rearrangement inequality**. -/
+lemma smul_add_smul_le_smul_add_smul' (ha : a₂ ≤ a₁) (hb : b₂ ≤ b₁) :
+    a₁ • b₂ + a₂ • b₁ ≤ a₁ • b₁ + a₂ • b₂ := by
+  simp_rw [add_comm (a₁ • _)]; exact smul_add_smul_le_smul_add_smul ha hb
+
+end PosSMulMono
+
+section PosSMulStrictMono
+variable [PosSMulStrictMono α β] {a₁ a₂ : α} {b₁ b₂ : β}
+
+/-- Binary strict **rearrangement inequality**. -/
+lemma smul_add_smul_lt_smul_add_smul (ha : a₁ < a₂) (hb : b₁ < b₂) :
+    a₁ • b₂ + a₂ • b₁ < a₁ • b₁ + a₂ • b₂ := by
+  obtain ⟨a, ha₀, rfl⟩ := lt_iff_exists_pos_add.1 ha
+  rw [add_smul, add_smul, add_left_comm]
+  gcongr
+  exact ha₀
```

Comment:
I'm slightly surprised `gcongr` can't figure this one out, it's just an assumption right?

Reply:
Yes but it's not in a position where `gcongr` would look for an assumption. Instead it's a position where `gcongr` calls `positivity`, but only once you import `Tactic.GCongr`. If you only import, `Tactic.GCongr.Core`, then it doesn't try to discharge.

Reply:
Then it sounds to me like `Tactic.GCongr` should be imported, since we'd like to discharge the positivity assumption!